### PR TITLE
[Filters] Make Filters Query Text Field Optional [#SOPS-81]

### PR DESCRIPTION
**Pull request by umar-k**
### WHY are these changes introduced?

Whenever a `Filters` component is used, it always includes the search text input field. 
Text search doesn't make sense in all contexts that a `Filters` can be used in.

Related Feature Request
https://github.com/Shopify/polaris-react/issues/3671

Original PR that was closed:
https://github.com/Shopify/polaris-react/pull/3674

### WHAT is this pull request doing?

This PR adds an optional boolean prop named `hideQueryField` which hides the query text field.

Here is the how the Filters component currently looks:
Current view of Filters Component

Here is what the Filters component looks with the query text field hidden:
Filters Component with the query field hidden

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

You can use any of the existing `Filters` examples and add the `hideQueryField` prop. 
This should hide the query filter text field.

**NOTE: I tried tophatting the documentation but I don't see my new example (`Filters with query field hidden`) showing as an example in the drop-down. I'm not sure if this is expected or if I'm doing something wrong.**

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.scss
Comment (kyledurand) [inline src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.scss:71]: Are these always rendered in a button group? I wonder if we could use our segmented data attribute here i.e.
```
.Item > [data-buttongroup-segmented='true'] {
  .Button {}
}
```

File: src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.scss
Comment (kyledurand) [inline src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.scss:87]: With all of this specificity do we still need `!important`?

File: src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.tsx
Comment (kyledurand) [inline src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.tsx:118]: Why do we need this?

File: src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.tsx
Comment (kyledurand) [inline src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.tsx:154]: Maybe add a test for this

**Review by kyledurand — COMMENTED** [review]
This is looking pretty good 👏

Just some minor comments

File: src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.scss
Comment (umar-k) [reply src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.scss:71]: Looks like we never actually use the Polaris `ButtonGroup` component in `Filters`.

We create individual `Button`s using the `activatorButtonFrom` function on `Line 234` in `ConnectedFilterControl.tsx` and then use CSS to override the border-radius / spacing

File: src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.scss
Comment (umar-k) [reply src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.scss]: Good suggestion! 
Had to change a few of the selectors but was able to remove all of the `!important` selectors 😄

File: src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.tsx
Comment (umar-k) [reply src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.tsx:118]: If the query field is hidden AND only the `More Filters` button is showing, we have to round the left borders.
Previously we didn't have to do that because the in the old design, the `More Filters` button would have been up against the text field.

**With Query Field Hidden:**
image



**Without Query Field Hidden:**
image

File: src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.scss
Comment (umar-k) [reply src/components/Filters/components/ConnectedFilterControl/ConnectedFilterControl.scss:70]: I think it would make sense to use a `ButtonGroup` so that we can delete a lot of the CSS in here.
I can try to tackle that in a future PR 👍

**Comment by BPScott** [comment]
I'm not a designer, but what if we keep the filter buttons aligned to the right? For some pulling them over to the left feels odd if the input element isn't there.

**Comment by umar-k** [comment]
> I'm not a designer, but what if we keep the filter buttons aligned to the right? For some pulling them over to the left feels odd if the input element isn't there.

@BPScott I talked about this with the Designer on our team. 

We were going to with with right-aligning the filters but decided to go with left-aligned because once you start applying filters, it looks like they're floating in a lot of empty space.
image

vs

image

**Comment by kyledurand** [comment]
Ben is back on Monday, can this wait for his final review?

**Comment by umar-k** [comment]
Yeah for sure 👍 
Sorry, didn't realize Ben was away

File: src/components/Filters/components/ConnectedFilterControl/tests/ConnectedFilterControl.test.tsx
Comment (BPScott) [inline src/components/Filters/components/ConnectedFilterControl/tests/ConnectedFilterControl.test.tsx:211]: One glorious day we'll complete the move from enzyme to `@shopify/react-testing` and we'll be able to write things like `expect(connectedFilterControl).not.toContainReactComponent('div', {className: 'CenterContainer'})` and it will be glorious.

Today is not that day though.

(Ignore this, nothing to do here, this is just me rambling)

**Review by BPScott — APPROVED** [review]
Looks all good to me :shipit: 

Some of that CSS looks brittle and scary but is basically the same as what we did before so that's not on you. Agree with the thoughts of revisiting that so it can leverage ButtonGroup (and probably enhancing ButtonGroup with `connectedLeft` / `connectedRight` props to match the existing `connectedTop`) but that's a story for another PR.

Thanks for the contribution, merge when you're ready!

**Comment by umar-k** [comment]
Thanks for the reviews @kyledurand @BPScott 👍 !!
Had to resolve merge conflict in `UNRELEASED.md`. Will merge soon as all tests pass again

**Pull request merged**
